File: Aquatrack-frontend/src/services/residentPaymentService.js
import api from "../api/axios";

// ==========================================
// Resident Payment Service
// ==========================================

// ==========================================
// Get My Payments
// ==========================================

export const getMyPayments = async () => {

    const response = await api.get(
        "/resident/payments"
    );

    return response.data;

};

// ==========================================
// Get My Payment By ID
// ==========================================

export const getMyPaymentById = async (
    paymentId
) => {

    const response = await api.get(
        `/resident/payments/${paymentId}`
    );

    return response.data;

};

// ==========================================
// Get My Payment Summary
// ==========================================

export const getMyPaymentSummary = async () => {

    const response = await api.get(
        "/resident/payments/summary"
    );

    return response.data;

};

// ==========================================
// Pay My Water Bill
// ==========================================

export const payMyWaterBill = async (
    paymentData
) => {

    const response = await api.post(
        "/resident/payments",
        paymentData
    );

    return response.data;

};

File: AquaTrack/src/main/java/com/aquatrack/service/ResidentPaymentService.java
package com.aquatrack.service;

import com.aquatrack.dto.payment.CreatePaymentRequestDto;
import com.aquatrack.dto.payment.PaymentResponseDto;
import com.aquatrack.dto.payment.PaymentSummaryResponseDto;

import java.util.List;

public interface ResidentPaymentService {

    // ==========================================
    // Get My Payments
    // ==========================================

    List<PaymentResponseDto> getMyPayments();

    // ==========================================
    // Get My Payment By ID
    // ==========================================

    PaymentResponseDto getMyPaymentById(
            Long paymentId
    );

    // ==========================================
    // Get My Payment Summary
    // ==========================================

    PaymentSummaryResponseDto getMyPaymentSummary();

    // ==========================================
    // Pay My Water Bill
    // ==========================================

    PaymentResponseDto payMyWaterBill(
            CreatePaymentRequestDto requestDto
    );

}
